### PR TITLE
Add version command and unit tests

### PR DESCRIPTION
**Pull request by lynnemorrison**
Add version command for kube/nonkube CLI.     The user can redirect
the output of this command to a file.  
Tested existing code that creates a manifest file and default manifest
with environmental variables.    Both work but there is no current CLI 
command to execute either of these features.

### CODE REVIEW

**Comment by nluaces** [comment]
This line 
https://github.com/skupperproject/skupper/blob/b2ebdcc683517af90265a8bb65b31df83be49d0f/Makefile#L132C2-L132C12

has to be changed with the new skupper version command (json format) that generates a file called `manifest.json`, like the old command. 

That make target is going to be used in the `config.yaml` configuration that runs in circle ci.

**Comment by lynnemorrison** [comment]
> This line https://github.com/skupperproject/skupper/blob/b2ebdcc683517af90265a8bb65b31df83be49d0f/Makefile#L132C2-L132C12
> 
> has to be changed with the new skupper version command (json format) that generates a file called `manifest.json`, like the old command.
> 
> That make target is going to be used in the `config.yaml` configuration that runs in circle ci.

@nluaces I updated the cmd/manifest/manifest.go to use the updated code.   Please let me know if this work for you.

**Comment by nluaces** [comment]
@lynnemorrison the manifest.go command needs to be removed. It is going to be replaced by `skupper version -o json`. There are a few things to do:

1. Remove file `cmd/manifest/manifest.go`
2. Replace the make target
```
generate-manifest: build-manifest
	./manifest
```

by these lines:

```
generate-manifest: build-cmd
	./skupper version -o json > manifest.json
	
```
you can test that everything is ok by running:

```
make clean
make generate-manifest
```

3. Remove these other lines
```
build-manifest:
	go build -ldflags="${LDFLAGS}"  -o manifest ./cmd/manifest
```

**Comment by nluaces** [comment]
There are other changes to do in `images.go`:

1) Remove these lines (and their dependencies):
```
	// These constants will be soon deprecated.
	ServiceControllerImageName string = "service-controller:main"
	FlowCollectorImageName     string = "flow-collector:main"
	SiteControllerImageName    string = "site-controller:main"
```
They are V1 images that we are not going to use in V2.

File: pkg/utils/configs/manifest.go [hunk 1]
```diff
@@ -42,7 +45,7 @@ func (manager *ManifestManager) GetDefaultManifestWithEnv() Manifest {
 	}
 }
 
-func (manager *ManifestManager) CreateFile(m Manifest) error {
+func (manager *ManifestManager) CreateFile(m SkupperManifest) error {
```
Comment (nluaces) [inline pkg/utils/configs/manifest.go:48]: I think this method can be removed, because no function is calling it.

File: pkg/images/images.go [hunk 2]
```diff
@@ -12,9 +12,10 @@ const (
 	PrometheusServerImageName string = "prometheus:v2.42.0"
 	OauthProxyImageRegistry   string = "quay.io/openshift"
 	OauthProxyImageName       string = "origin-oauth-proxy:4.14.0"
+)
```
Comment (nluaces) [inline pkg/images/images.go:15]: `NetworkConsoleCollector` is now `NetworkObserver`

So this line:
```
NetworkConsoleCollectorImageName string = "network-console-collector:v2-latest"
```

should be changed to
```
NetworkObserverImageName string = "network-observer:v2-latest"
```

File: internal/cmd/skupper/version/nonkube/version.go [hunk 3]
```diff
@@ -20,13 +34,56 @@ func NewCmdVersion() *CmdVersion {
...
 
-func (cmd *CmdVersion) InputToOptions() {}
+func (cmd *CmdVersion) Run() error {
+	files := cmd.manifest.GetConfiguredManifest()
+	if cmd.output != "" {
+		encodedOutput, err := utils.Encode(cmd.output, files)
+		if err != nil {
+			return err
+		}
+		fmt.Println(string(encodedOutput))
```
Comment (nluaces) [inline internal/cmd/skupper/version/nonkube/version.go:76]: the `string` method can be removed

File: internal/cmd/skupper/version/kube/version.go [hunk 4]
```diff
@@ -28,16 +37,53 @@ func (cmd *CmdVersion) NewClient(cobraCommand *cobra.Command, args []string) {
...
+}
+
+func (cmd *CmdVersion) Run() error {
+	files := cmd.manifest.GetConfiguredManifest()
+	if cmd.output != "" {
+		encodedOutput, err := utils.Encode(cmd.output, files)
+		if err != nil {
+			return err
+		}
+		fmt.Println(string(encodedOutput))
```
Comment (nluaces) [inline internal/cmd/skupper/version/kube/version.go:76]: the `string` method can be removed

Comment (lynnemorrison) [reply internal/cmd/skupper/version/kube/version.go]: done

File: internal/cmd/skupper/version/nonkube/version.go (hunk 3 above)
Comment (lynnemorrison) [reply internal/cmd/skupper/version/nonkube/version.go]: done

File: pkg/images/images.go (hunk 2 above)
Comment (lynnemorrison) [reply pkg/images/images.go:15]: done

File: pkg/utils/configs/manifest.go (hunk 1 above)
Comment (lynnemorrison) [reply pkg/utils/configs/manifest.go]: done

**Review by nluaces — APPROVED** [review]
Looks good to me!

**Pull request merged**